### PR TITLE
Form violations update and minor changes

### PR DESCRIPTION
**Pull request by bedakb**

### CODE REVIEW

File: src/dialog/dialog.tsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ class Dialog extends React.Component<Props, {}> {
                     onKeyUp={this.onKeyupHandler}
                     tabIndex={-1}
                     role="dialog"
-                    data-qa={`dialog-${this.props.id}`}
+                    data-qa={`dialog`}
```
Comment (BenjaminEckardt) [inline src/dialog/dialog.tsx]: Why was it removed?

File: src/form/form.tsx [hunk 2]
```diff
@@ -308,6 +308,9 @@ class Form<MODEL extends object> extends React.Component<Props<MODEL>, State<MOD
         return getDeclaredFields(this.props.children)
             .map((field: Field<any, MODEL>) => field.id)
             .map(this.getFieldViolations)
+            .filter((violations: Array<ConstraintViolation>) =>
+                violations.every((violation) => violation.severity === ViolationSeverity.ERROR)
```
Comment (BenjaminEckardt) [inline src/form/form.tsx]: Doesn't that return true if you have warning and error for a particular field?

Comment (bedakb) [reply src/form/form.tsx]: It will return `false` If you have for e.g `[INFO, ERROR]`.

File: src/dialog/dialog.tsx (hunk 1 above)
Comment (BenjaminEckardt) [reply src/dialog/dialog.tsx]: Clarified in person

File: src/form/form.tsx (hunk 2 above)
Comment (BenjaminEckardt) [reply src/form/form.tsx]: Clarified in person

**Pull request merged**